src/Form/ChoiceList/ModelChoiceLoader.php

```diff
@@ -99,26 +98,26 @@ public function loadChoiceList($value = null): ChoiceListInterface
                     // Otherwise expect a __toString() method in the entity
                     $valueObject = (string) $model;
                 } else {
-                    $class = $this->modelManager instanceof ProxyResolverInterface
-                        ? $this->modelManager->getRealClass($model)
-                        // NEXT_MAJOR: Change to `\get_class`
-                        : BCHelper::getClass($model);
-
-                    throw new \LogicException(sprintf(
+                    throw new TransformationFailedException(sprintf(
                         'Unable to convert the model "%s" to string, provide "property" option'
                         .' or implement "__toString()" method in your model.',
-                        $class
+                        $this->class
                     ));
                 }
 
                 if (!\array_key_exists($valueObject, $choices)) {
                     $choices[$valueObject] = [];
                 }
 
-                $choices[$valueObject][] = implode(
```

Comment:
getNormalizedIdentifier is already doing the implode
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/4.x/src/Model/ModelManager.php#L293